eventhub_endpoint.bal

```diff
@@ -87,29 +84,17 @@ public client class Client {
         }
     }
 
-    # Get request with common headers
-    #
-    # + return - Return a http request with authorization header
-    private isolated function getAuthorizedRequest() returns http:Request {
-        http:Request req = new;
-        req.addHeader("Authorization", self.getSASToken());
-        if (!self.config.enableRetry) {
-            // disable automatic retry
-            req.addHeader("x-ms-retrypolicy", "NoRetry");
-        }
-        return req;
-    }
-
     # Send batch of events
     #
-    # + batchEvent - batch of events
     # + eventHubPath - event hub path
+    # + batchEvent - batch of events
     # + partitionId - partition ID
     # + publisherId - publisher ID
     # + return - Eventhub error if unsuccessful
-    public remote function sendBatch(string eventHubPath, BatchEvent batchEvent, int partitionId = -1, string publisherId = "") returns @tainted error? {
-        http:Request req = self.getAuthorizedRequest();
-        req.setJsonPayload(self.getBatchEventJson(batchEvent));
+    remote function sendBatch(string eventHubPath, BatchEvent batchEvent, int partitionId = -1, string publisherId = "") 
+        returns @tainted error? {
+        http:Request req = getAuthorizedRequest(self.config);
+        req.setJsonPayload(getBatchEventJson(batchEvent));
         req.setHeader("content-type", "application/vnd.microsoft.servicebus.json");
```

Comment:
Ballerina HTTP module API doc has defined constants for common headers. You can use them in all such cases.

Reply:
Using predefined constants via setContentType() require variable assignment. Need to analyse the better approach.